runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -0,0 +1,230 @@
+     * @param corfuRuntime            Connected instance of the Corfu Runtime.
+     * @param serializer              Protobuf Serializer.
+     */
+    @Nonnull
+    public Table(@Nonnull final String namespace,
+                 @Nonnull final String fullyQualifiedTableName,
+                 @Nonnull final V valueSchema,
+                 @Nullable final M metadataSchema,
+                 @Nonnull final CorfuRuntime corfuRuntime,
+                 @Nonnull final ISerializer serializer) {
+
+        this.corfuRuntime = corfuRuntime;
+        this.namespace = namespace;
+        this.fullyQualifiedTableName = fullyQualifiedTableName;
+        this.metadataOptions = Optional.ofNullable(metadataSchema)
+                .map(schema -> MetadataOptions.builder()
+                        .metadataEnabled(true)
+                        .defaultMetadataInstance(schema)
+                        .build())
+                .orElse(MetadataOptions.builder().build());
+
+        this.corfuTable = corfuRuntime.getObjectsView().build()
+                .setTypeToken(CorfuTable.<K, CorfuRecord<V, M>>getTableType())
+                .setStreamName(this.fullyQualifiedTableName)
+                .setSerializer(serializer)
+                .setArguments(new ProtobufIndexer(valueSchema))
+                .open();
+    }
+
+    /**
+     * Returns the underlying CorfuTable object.
+     * NOTE: For internal use only.
+     *
+     * @return CorfuTable instance.
+     */
+    public CorfuTable<K, CorfuRecord<V, M>> getUnderlyingObject() {
+        return corfuTable;
+    }
+
+    /**
+     * Begins a write after write transaction.
+     */
+    private boolean TxBegin() {
+        if (TransactionalContext.isInTransaction()) {
+            return false;
+        }
+        corfuRuntime.getObjectsView()
+                .TXBuild()
+                .type(TransactionType.OPTIMISTIC)
+                .build()
+                .begin();
+        return true;
+    }
+
+    /**
+     * Ends an ongoing transaction.
+     */
+    private void TxEnd() {
+        corfuRuntime.getObjectsView().TXEnd();
+    }
+
+    /**
+     * Create a new record.
+     *
+     * @param key      Key
+     * @param value    Value
+     * @param metadata Record metadata.
+     * @return Previously stored record if any.
+     */
+    @Nullable
+    CorfuRecord<V, M> create(@Nonnull final K key,
+                             @Nullable final V value,
+                             @Nullable final M metadata) {
+        boolean beganNewTxn = false;
+        try {
+            beganNewTxn = TxBegin();
+            return corfuTable.put(key, new CorfuRecord<>(value, metadata));
+        } finally {
+            if (beganNewTxn) {
+                TxEnd();
+            }
+        }
+    }
+
+    /**
+     * Fetch the value for a key.
+     *
+     * @param key Key.
+     * @return Corfu Record for key.
+     */
+    @Nullable
+    public CorfuRecord<V, M> get(@Nonnull final K key) {
```

Comment:
Do we support snapshot read? Is that introduced in part 3 patch?

Reply:
Yes.